src/query/storages/fuse/src/operations/recluster.rs

```diff
@@ -156,7 +158,7 @@ impl FuseTable {
         }
 
         // sort
-        let block_size = ctx.get_settings().get_max_block_size()? as usize;
+        let block_size = self.get_option(FUSE_OPT_KEY_ROW_PER_BLOCK, DEFAULT_BLOCK_MAX_ROWS);
```

Comment:
oh, maybe `settings.max_block_size` is better?

value of table options  FUSE_OPT_KEY_ROW_PER_BLOCK may be much larger than settings.max_block_size.

Reply:
I think `settings.max_block_size` is better. The block size threshold is used to split large block in Sort.